### PR TITLE
Make module register users & sign_in & sign_out

### PR DESCRIPTION
**Pull request by l3trungkjen**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,13 +1,19 @@
 class User < ActiveRecord::Base
+  ADMIN = %w(0 1)
```
Comment (dieunb) [inline app/models/user.rb]: Name should `ROLE`

File: config/database.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# SQLite version 3.x
```
Comment (dieunb) [inline config/database.yml]: remove this file from this pull

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -2,4 +2,10 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.for(:sign_up) {|u| u.permit :role, :name, :email, :password, :password_confirmation}
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: line too length

File: app/models/user.rb [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
+
+  before_save :set_default
+
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+
+  private
+  def set_default
+    self.email = email.downcase
+    self.role = ADMIN.first
```
Comment (dieunb) [inline app/models/user.rb]: ???

Comment (l3trungkjen) [inline app/models/user.rb]: quên :D

**Pull request merged**